infrastructure/router/router.go

```diff
@@ -0,0 +1,35 @@
+package router
+
+import (
+	"log"
+
+	"github.com/wmaldonadoc/academy-go-q42021/config"
+	"github.com/wmaldonadoc/academy-go-q42021/interface/controller"
+
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+)
+
+// NewRouter - Setup & returns the API routes.
+func NewRouter(controller controller.AppController) {
+	port := config.GetEnvVariable("PORT")
+	router := gin.Default()
+	mainGroup := router.Group("/api/v1")
+	{
+		pokemonRoutes := mainGroup.Group("/pokemons")
+		{
+			pokemonRoutes.GET("/:id", func(context *gin.Context) { controller.Pokemon.GetByID(context) })
+			pokemonRoutes.GET("/name/:name", func(context *gin.Context) { controller.Pokemon.GetByName(context) })
+		}
+
+		healthRoutes := mainGroup.Group("/health")
+		{
+			healthRoutes.GET("/", func(context *gin.Context) { controller.Health.GetServiceHealth(context) })
+		}
+	}
+	if err := router.Run(port); err == nil {
+		zap.S().Infof("API running at port", port)
+	} else {
+		log.Fatal("Error running router")
+	}
```

Comment:
Should never use Fatal; main should be the only one to do so. Being so, this method should return an error, so `main.go` could handle that and execute Fatal if it's necessary.

Again, let's align with error handling best practices.
```suggestion
	}
	
	if err := router.Run(port); err != nil {
		log.Fatal("Error running router")
	} 
	
	zap.S().Infof("API running at port", port)
	}
```